### ISSUE TITLE
PdfReadError: Could not find xref table at specified location

### ISSUE THREAD

**Issue by MartinThoma** [py-pdf/PyPDF2#1089]
I wanted to get metadata from a PDF. The PDF opens fine in Chrome.

## Environment

Which environment were you using when you encountered the problem?

```bash
$ python -m platform
Linux-5.4.0-121-generic-x86_64-with-glibc2.31

$ python -c "import PyPDF2;print(PyPDF2.__version__)"
2.4.2
```

## Code + PDF

The PDF: [`pdf/6f6d505d151b9769900a8bd315db6e67.pdf`](https://corpora.tika.apache.org/base/docs/govdocs1/934/934771.pdf)

```pycon
>>> from PyPDF2 import PdfReader
>>> reader = PdfReader('pdf/6f6d505d151b9769900a8bd315db6e67.pdf')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/moose/Github/py-pdf/PyPDF2/PyPDF2/_reader.py", line 267, in __init__
    self.read(stream)
  File "/home/moose/Github/py-pdf/PyPDF2/PyPDF2/_reader.py", line 1305, in read
    raise PdfReadError("Could not find xref table at specified location")
PyPDF2.errors.PdfReadError: Could not find xref table at specified location
```

**Comment by MartinThoma** [comment]
Maybe https://github.com/py-pdf/PyPDF2/pull/493 helps?

**Comment by pubpub-zz** [comment]
the problem is that the /Prev in the 1st chained Trailer is pointing to an invalid position. In such case the xref shall be completely rebuild parsing the whole file
proposed fixed in #1133
